Docs: template parameters in KATANA_CHECKED

scober opened the pull request:
Document a work-around for using KATANA_CHECKED and
KATANA_CHECKED_CONTEXT with functions that accept multiple template
parameters.

danielmawhirter commented:
LGTM

danielmawhirter reviewed (approved):
LGTM (but as a review)

Commit:
Docs: template parameters in KATANA_CHECKED

Document a work-around for using KATANA_CHECKED and
KATANA_CHECKED_CONTEXT with functions that accept multiple template
parameters.

ddn0 commented on libsupport/include/katana/Result.h:
```diff
@@ -639,6 +639,16 @@ CheckedExpressionToValue(arrow::Status&&) {
+/// are not enclosed in a pair of parentheses as delimiters for its own
+/// arguments. This is problematic for functions with multiple template
+/// parameters. The upshot of this is that
+/// ``KATANA_CHECKED_CONTEXT(MakeSword<Handle, Blade>(), "third argument")``
```
There are multiple levels of markup interpretation Doxygen -> Sphinx (RST).

If you want to show a code block in Doxygen, you can follow the example in this file and insert a new line and then prefix with 4 spaces.

scober replied:
I changed to using Doxygen style inline code formatting. If I understand correctly, Doxygen will parse that to a Doxygen representation of inline code and rst will respect that. So it should render as code in both sets of docs?

ddn0 replied:
I personally prefer a code block (new line, prefix with 4 spaces) for this as it is easier to read in raw form and in a rendered document.

And for your example, it would probably be confusing if the example were wrapped among two lines.
